### PR TITLE
Feature: tax info page 

### LINKED ISSUE #160: Tax info page

### PR DESCRIPTION
**Pull request by rpollock03**
Resolves #160 - Add tax info pages and new country select dropdown component
Screenshot 2022-12-22 at 10 46 32

- checks firebase for existing user tax info and pre-populates form with relevant data
- updates firebase with new data if all fields are populated.

Screenshot 2022-12-22 at 10 52 34

Screenshot 2022-12-22 at 10 41 15

### CODE REVIEW

**Comment by rpollock03** [comment]
resolves #160

File: components/CountrySelectDropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,482 @@
...
+
+export function CountrySelect(props) {
+  return (
+    <Autocomplete
+      sx={{ width: 300 }}
+      options={countries}
+      autoHighlight
+      size="small"
+      onInputChange={(e, newValue) => props.handleCountrySelect(newValue)}
+      /* defaultValue={{
```
Comment (piotrostr) [inline components/CountrySelectDropdown.tsx:14]: no stranded comments

File: components/CountrySelectDropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,482 @@
...
+          component="li"
+          sx={{ "& > img": { mr: 2, flexShrink: 0 } }}
+          {...props}
+        >
+          <img
+            loading="lazy"
+            width="20"
+            src={`https://flagcdn.com/w20/${option.code.toLowerCase()}.png`}
+            srcSet={`https://flagcdn.com/w40/${option.code.toLowerCase()}.png 2x`}
+            alt=""
```
Comment (piotrostr) [inline components/CountrySelectDropdown.tsx:30]: always include image `alt`

File: components/CountrySelectDropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,482 @@
...
+        label: 'United Kingdom',
+      }}*/
+      inputValue={props.country}
+      renderOption={(props, option) => (
+        <Box
+          component="li"
+          sx={{ "& > img": { mr: 2, flexShrink: 0 } }}
+          {...props}
+        >
+          <img
```
Comment (piotrostr) [inline components/CountrySelectDropdown.tsx:25]: don't use the vanilla image, use `Image` from nextjs

File: components/CountrySelectDropdown.tsx [hunk 4]
```diff
@@ -0,0 +1,482 @@
+import * as React from "react"
+import Box from "@mui/material/Box"
+import TextField from "@mui/material/TextField"
+import Autocomplete from "@mui/material/Autocomplete"
+
+export function CountrySelect(props) {
```
Comment (piotrostr) [inline components/CountrySelectDropdown.tsx:6]: no props type specified, please add an interface and use de-structuring as well

File: pages/taxinfo.tsx [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+  const firestore = useFirestore()
+  const heroRef = doc(firestore, `heroes/${user?.uid}`)
+  const { status: heroStatus, data: hero } = useFirestoreDocData(heroRef)
+
+  const [address, setAddress] = useState("")
+  const [city, setCity] = useState("")
+  const [postcode, setPostcode] = useState("")
+  const [country, setCountry] = useState("")
+
+  useEffect(() => {
```
Comment (piotrostr) [inline pages/taxinfo.tsx:27]: this might not be safe, make sure the properties are there, if `hero.location` is undefined this will throw an error

**Review by piotrostr — CHANGES_REQUESTED** [review]
looking good overall, some slight changes to be introduced

**Comment by rpollock03** [comment]
> looking good overall, some slight changes to be introduced

Thanks, requested changes done.

File: pages/taxinfo.tsx [hunk 6]
```diff
@@ -25,7 +25,7 @@ export default function Index() {
   const [country, setCountry] = useState("")
 
   useEffect(() => {
-    if (heroStatus == "success") {
+    if (heroStatus == "success" && typeof hero.location != "undefined") {
```
Comment (piotrostr) [inline pages/taxinfo.tsx:28]: would rather use null chaining or just ensuring that if the data is not there the UI is not going to break

please ensure that prior to merging, I'll LGTM still

**Review by piotrostr — APPROVED** [review]

**Review by piotrostr — APPROVED** [review]

**Comment by rpollock03** [comment]
Merged via a new PR due to conflicts in the bun lock files.

**Pull request closed without merging**